### PR TITLE
StackHCIVM 2024-01-01 api version 

### PR DESCRIPTION
**Pull request by hvedati**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/StackHCIVM/StackHCIVM.Autorest/Az.StackHCIVM.psd1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @{
   GUID = '7389f67c-ec9d-4c13-9c9a-308834413af6'
   RootModule = './Az.StackHCIVM.psm1'
-  ModuleVersion = '0.1.0'
+  ModuleVersion = '2.0.0'
```
Comment (vidai-msft) [inline src/StackHCIVM/StackHCIVM.Autorest/Az.StackHCIVM.psd1:4]: Do NOT update the module version here.

File: src/StackHCIVM/StackHCIVM.Autorest/README.md [hunk 2]
```diff
@@ -54,23 +54,23 @@ In this directory, run AutoRest:
...
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/marketplaceGalleryImages.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/networkInterfaces.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/storageContainers.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/virtualHardDisks.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/virtualMachineInstances.json
+  - $(repo)/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/logicalNetworks.json
   - $(repo)/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-03-15-preview/HybridCompute.json
 
-module-version: 0.1.0
+module-version: 2.0.0
```
Comment (vidai-msft) [inline src/StackHCIVM/StackHCIVM.Autorest/README.md:73]: Do NOT update the version

**Review by vidai-msft — CHANGES_REQUESTED** [review]
Please fix the test errors on CI.

**Comment by vidai-msft** [comment]
@hvedati The build issue has been resolved. Please fix the test issue.

**Comment by vidai-msft** [comment]
@hvedati Can you please select the target release for this PR in the Mandatory Checklist section?

**Comment by vidai-msft** [comment]
@hvedati Please leave a comment here as the change log message.

**Comment by hvedati** [comment]
Updating API to 2024-01-01 version.

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**